test/eqc_util.erl

```diff
@@ -0,0 +1,173 @@
+         riak_object:set_vclock(riak_object:new(Bucket, Key, Value),
+                                Vclock)).
+
+not_empty(G) ->
+    ?SUCHTHAT(X, G, X /= [] andalso X /= <<>>).
+
+bkey() ->
+    {non_blank_string(),  %% bucket
+     non_blank_string()}. %% key
+
+non_blank_string() ->
+    ?LET(X,not_empty(list(lower_char())), list_to_binary(X)).
+
+%% Generate a lower 7-bit ACSII character that should not cause any problems
+%% with utf8 conversion.
+lower_char() ->
+    choose(16#20, 16#7f).
+
+%%%===================================================================
+%%% Mocks, Env-Sets
+%%%===================================================================
+
+set_core_envs() ->
+    set_core_envs([]).
+
+set_core_envs(Options) ->
+    HashFun = proplists:get_value(chash_keyfun, Options,
+                                  {riak_core_util,chash_std_keyfun}),
+    application:set_env(riak_core, default_bucket_props,
+                        [{n_val, 1},
+                         {search_index, "test"},
+                         {chash_keyfun,HashFun}]),
+    application:set_env(riak_core, ring_state_dir, "ring"),
+    application:set_env(riak_core, platform_data_dir, "eqc_test_data"),
+    application:set_env(riak_core, handoff_port, 0).
+
+set_kv_envs() ->
+    application:set_env(riak_kv, anti_entropy_data_dir, "kv_aae"),
+    application:set_env(riak_kv, anti_entropy, {on, []}).
+
+set_yokozuna_envs() ->
+    application:set_env(yokozuna, anti_entropy_data_dir, "yz_aae"),
+    application:set_env(yokozuna, enabled, true).
+
+start_mock_components() ->
+    start_mock_components([]).
+
+start_mock_components(Options) ->
+    RingSize = proplists:get_value(ring_size, Options, 64),
+    application:load(riak_core),
+    application:load(riak_kv),
+    application:load(lager),
+    ok = lager:start(),
+    riak_core_metadata_manager:start_link([{data_dir, "eqc_test_data"}]),
+    riak_core_ring_events:start_link(),
+    riak_core_ring_manager:start_link(test),
+    riak_core_ring_manager:setup_ets(test),
+    setup_mockring(RingSize),
+    riak_kv_entropy_info:create_table(),
+    riak_kv_entropy_manager:start_link(),
+    ok.
+
+cleanup_mock_components() ->
+    riak_core_ring_manager:cleanup_ets(test),
+    application:stop(lager),
+    application:unload(lager),
+    stop_process(riak_core_ring_manager),
+    stop_process(riak_core_metadata_manager),
+    stop_process(riak_kv_entropy_manager),
+    application:stop(riak_core),
+    application:stop(riak_kv),
+    application:unload(riak_core),
+    application:unload(riak_kv),
+    ok.
+
+%% Stop a running pid - unlink and exit(kill) the process
+stop_process(undefined) ->
+    ok;
+stop_process(RegName) when is_atom(RegName) ->
+    stop_process(whereis(RegName));
+stop_process(Pid) when is_pid(Pid) ->
+    unlink(Pid),
+    exit(Pid, shutdown),
+    ok = wait_for_pid(Pid).
+
+%% Wait for a pid to exit
+wait_for_pid(Pid) ->
+    Mref = erlang:monitor(process, Pid),
```

Comment:
There's an inherent race in this code. You should monitor before sending the exit message.